util/chunk/row_container.go

```diff
@@ -239,17 +282,181 @@ func (a *SpillDiskAction) SetFallback(fallback memory.ActionOnExceed) {
 // SetLogHook sets the hook, it does nothing just to form the memory.ActionOnExceed interface.
 func (a *SpillDiskAction) SetLogHook(hook func(uint64)) {}
 
-// ResetOnce resets the spill action so that it can be triggered next time.
-func (a *SpillDiskAction) ResetOnce() {
+// ResetRowContainer resets the spill action and sets the RowContainer for the SpillDiskAction.
+func (a *SpillDiskAction) ResetRowContainer(c *RowContainer) {
 	a.m.Lock()
 	defer a.m.Unlock()
-	a.once = sync.Once{}
+	a.c = c
+}
+
+// ErrInsertToPartitionFailed indicate that the SortedRowContainer is sorted and prohibit inserting data.
+var ErrInsertToPartitionFailed = terror.ClassExecutor.New(1, "Insert the records unsuccessfully. The SortedRowContainer is sorted.")
```

Comment:
The word `Partition` is used in sort executor, but it's confused for code readers when reading this package. I think `ErrCannotAddBecauseSorted` would be better. Besides, a simple `errors.New` may be more suitable.

```suggestion
var ErrCannotAddBecauseSorted = errors.New("can not add because sorted")
```